src/Distributed/halo_communication.jl

```diff
@@ -28,46 +23,56 @@ import Oceananigans.BoundaryConditions:
     fill_south_and_north_halo!,
     fill_bottom_and_top_halo!
 
-@inline sync_device!(::GPU) = synchronize()
-
 #####
 ##### MPI tags for halo communication BCs
 #####
 
-sides  = (:west, :east, :south, :north, :top, :bottom)
-side_id = Dict(side => n for (n, side) in enumerate(sides))
+sides  = (:west, :east, :south, :north, :southwest, :southeast, :northwest, :northeast)
+side_id = Dict(side => n-1 for (n, side) in enumerate(sides))
 
 opposite_side = Dict(
-    :west => :east, :east => :west,
-    :south => :north, :north => :south,
-    :bottom => :top, :top => :bottom
+    :west => :east, 
+    :east => :west,
+    :south => :north,
+    :north => :south,
+    :southwest => :northeast, 
+    :southeast => :northwest, 
+    :northwest => :southeast, 
+    :northeast => :southwest, 
```

Comment:
Why do corners communicate?

Reply:
in case of 2D partitioning, to avoid double pass.